FAQ.md

```diff
@@ -203,3 +203,71 @@ targets:
 ```
 
 Also, you can also check out the example configuration in the Mockito repository. 
+
+
+## How do I mock a `sealed` class?
+
+Suppose that you have code such as:
+
+```dart
+class Bar {
+  int value;
+
+  Bar(this.value);
+
+  Bar.zero() : value = 0;
+}
+
+class Foo {
+  Bar someMethod(int value) => Bar(value);
+}
+```
+
+and now you want to mock `Foo`. The generated implementation for `MockFoo`
+needs to return *something* if `someMethod` is called. It can't return `null`
+since its return type is non-nullable. It can't construct a `Bar` on its own
+without understanding the semantics of `Bar`'s constructors. That is, which
+`Bar` constructor should be called and with what arguments? To avoid this,
+Mockito instead generates its own, fake implementation of `Bar` that it does
+know how to construct, something like:
+
+```dart
+class _FakeBar extends Fake implements Bar {}
+```
+
+And then the generated implementation of `MockFoo` can have its `someMethod`
```

Comment:
It's not really an "implementation" of `MockFoo`; you could say the generated _declaration_ of `MockFoo` or maybe the generated mock implementation of `Foo`.

Reply:
Argh, yeah, I didn't mean "implementation" in the Dart `implements` sense.  I will just remove "the generated implementation of".